lib/reversetunnel/localsite.go

```diff
@@ -497,10 +497,115 @@ func (s *localSite) periodicFunctions() {
 			if err := s.sshTunnelStats(); err != nil {
 				s.log.Warningf("Failed to report SSH tunnel statistics for: %v: %v.", s.domainName, err)
 			}
+			s.agentStats()
 		}
 	}
 }
 
+// agentStats leverages the accessPoint cache to report all agents/proxies/auth servers connected to the
+// teleport cluster via prometheus metrics
+func (s *localSite) agentStats() {
+	hostID := make(map[string]struct{})
+	versionCount := make(map[string]int)
+
+	// Nodes, Proxies, Auths, KubeServices, and WindowsDesktopServices use the UUID as the name field where
+	// DB and App store it in the spec. Check expiry due to DynamoDB taking up to 48hr to expire from backend
+	// and then store hostID and version count information.
+	serverCheck := func(server interface{}) {
+		type serverKubeWindows interface {
+			Expiry() time.Time
+			GetName() string
+			GetTeleportVersion() string
+		}
+		type appDB interface {
+			serverKubeWindows
+			GetHostID() string
+		}
+
+		// appDB needs to be first as it also matches the serverKubeWindows interface
+		switch t := server.(type) {
+		case appDB:
```

Comment:
This works!? I'm surprised, for two reasons. 

1. I thought the cases had to be concrete types, not other interfaces
2. `appDB` implements `serverKubeWindows` via embedding

The spec has this to say:

> Cases then match actual types T against the dynamic type of the expression x. As with type assertions, x must be of interface type, and each non-interface type T listed in a case must implement the type of x. The types listed in the cases of a type switch must all be different.

This seems to imply that the cases should be non-interface types.

I would rewrite it just to be safe.

```
if a, ok := server.(appDB); ok {

} else if s, ok := server.(serverKubeWindows); ok {

}
```